### PR TITLE
Added sphinx/readthedocs documentation

### PR DESCRIPTION
**Pull request by isaac-tfn**
Describe Changes
---

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [x] Have new dependencies been added?
  - [x] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [x] Have they been added to `setup.cfg`

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import caf.distribute
+
+dir_path = Path(__file__).parents[1]
+source = dir_path / "src" / "caf" / "distribute"
+
+sys.path.insert(0, os.path.abspath(str(source)))
+sys.path.insert(
+    0,
+    os.path.abspath(r"C:\Users\IsaacScott\anaconda3\envs\distribute\Lib\site-packages"),
+)
```
Comment (BenTaylor-TfN) [inline docs/conf.py:31]: Is this path just used for local builds? 
We seem to be able to publish to ReadTheDocs, so I would guess so.

Can you make it relative so anyone can build local docs?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+project = "caf.distribute"
+copyright = "2023, Benjamin Taylor"
+author = "Benjamin Taylor"
```
Comment (BenTaylor-TfN) [inline docs/conf.py:35]:
```suggestion
copyright = "2023, Transport for the North"
author = "Transport for the North"
```

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+napoleon_google_docstring = False
+napoleon_numpy_docstring = True
+
+version = str(caf.distribute.__version__)
+# The full version, including alpha/beta/rc tags
+release = version
```
Comment (BenTaylor-TfN) [inline docs/conf.py:42]: Is version used later, or can you just assign it to release straight away?

File: docs/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Required libraries
+
+# Example
```
Comment (BenTaylor-TfN) [inline docs/requirements.txt:3]:
```suggestion
# Required libraries for a doc build
```

File: docs/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+scipy>=1.9.3
+numpy>=1.19.0
+pandas>=1.4.0
+tqdm>=4.50.2
+caf.toolkit>=0.2.1
+sphinx-automodapi>=0.16.0
+pydata-sphinx-theme>=0.14.1
+caf.distribute>=0.1.0
+graphviz>=0.20.1
```
Comment (BenTaylor-TfN) [inline docs/requirements.txt:12]: Can this be as simple as:
```suggestion
caf.distribute>=0.1.0
sphinx-automodapi>=0.16.0
pydata-sphinx-theme>=0.14.1
graphviz>=0.20.1
```

Making a few assumptions here:
- These requirements are just to build the docs
- only distribute is really needed so that it can get the version

File: docs/index.rst [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+Welcome to caf.distribute's documentation!
+=======================================
+
+A module of CAF for distributing demand across matrices.
+
+The Common Analytical Framework (CAF) is a toolkit of transport planning and appraisal functionalities. It's part of a project to make a lot of the useful transport related functionality more widely available and easily accessible. Other modules include:
+
+* `CAF.Toolkit <https://caftoolkit.readthedocs.io/en/latest/>`_
+
+* `CAF.Space <https://cafspcae.readthedocs.io/en/latest/>`_
```
Comment (BenTaylor-TfN) [inline docs/index.rst:15]:
```suggestion
* `CAF.Space <https://cafspace.readthedocs.io/en/latest/>`_
```

File: docs/index.rst [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+   sphinx-quickstart on Wed Oct  4 13:40:35 2023.
+   You can adapt this file completely to your liking, but it should at least
+   contain the root `toctree` directive.
+
+Welcome to caf.distribute's documentation!
+=======================================
+
+A module of CAF for distributing demand across matrices.
+
+The Common Analytical Framework (CAF) is a toolkit of transport planning and appraisal functionalities. It's part of a project to make a lot of the useful transport related functionality more widely available and easily accessible. Other modules include:
```
Comment (BenTaylor-TfN) [inline docs/index.rst:11]:
```suggestion
The Common Analytical Framework (CAF) is a collection of transport planning and appraisal functionalities. It's part of a project to make useful transport related functionality more widely available and easily accessible. Other modules include:
```

File: docs/index.rst [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+* `CAF.Toolkit <https://caftoolkit.readthedocs.io/en/latest/>`_
+
+* `CAF.Space <https://cafspcae.readthedocs.io/en/latest/>`_
+
+Tool info
+---------
+
+{Add any necessary tool info here Ben}
```
Comment (BenTaylor-TfN) [inline docs/index.rst:20]: You might need to check my syntax here as I was guessing based on the stuff above.

```suggestion
CAF.distribute focusses on tools and models to distribute vectors of data into matrices.
Currently, it only contains a self calibrating gravity model that is capable of calibrating data to a single cost distribution area.

Future plans involve enhancing this gravity model to handle multiple cost distribution areas, that is if the gravity model data covered two counties, a different cost distribution could be used for each county, while ensuring all totals still match.
There are also plans to move the `Iterative Proportional Fitting <https://en.wikipedia.org/wiki/Iterative_proportional_fitting>`_ algorithm over that currently exists in `CAF.Toolkit <https://github.com/Transport-for-the-North/caf.toolkit/blob/3c7ba4b7d770e90fcadb81672c9c485bcf08cecf/src/caf/toolkit/iterative_proportional_fitting.py#L837>`_.
```

**Review by BenTaylor-TfN — CHANGES_REQUESTED** [review]
Few minor changes to make, otherwise looks good!

File: docs/conf.py (hunk 3 above)
Comment (isaac-tfn) [reply docs/conf.py:42]: It's not - I just copied the pandas snippet you sent

File: docs/requirements.txt (hunk 5 above)
Comment (isaac-tfn) [reply docs/requirements.txt:10]: All are needed for the sphinx build github action. Although all other requirements are installed by installing distribute, I think it's better practice to keep them all in explicitly but it's your call.

Comment (BenTaylor-TfN) [reply docs/requirements.txt:10]: Do you know how the sphinx build installs distribute? Is it a local install?

If so, it will just use the already defined requirements.txt. At least this way we prevent repeating ourselves - we are only copying that file into here after all.

Comment (isaac-tfn) [reply docs/requirements.txt:10]: I don't think it's local no, it's from pypi, so as an edge case it could install a previous version with different requirements and then raise an error when it tries to import something which isn't installed. It isn't an issue if the version of distribute is out of date because it just needs to be able to import it without an error message. As I think about it that same error could occur on readthedocs too.

File: docs/conf.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+project = "caf.distribute"
+copyright = "2023, Transport for the North"
+author = "Transport for the North"
+
+napoleon_google_docstring = False
+napoleon_numpy_docstring = True
+
+version = str(caf.distribute.__version__)
+# The full version, including alpha/beta/rc tags
+release = version
```
Comment (BenTaylor-TfN) [inline docs/conf.py:42]:
```suggestion
# The full version, including alpha/beta/rc tags
release = str(caf.distribute.__version__)
```

**Review by BenTaylor-TfN — APPROVED** [review]
